src/main/java/net/minecraftforge/server/command/HelpCommand.java

```diff
@@ -0,0 +1,125 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package net.minecraftforge.server.command;
+
+import net.minecraft.command.CommandBase;
+import net.minecraft.command.CommandException;
+import net.minecraft.command.ICommand;
+import net.minecraft.command.ICommandSender;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.util.text.TextComponentTranslation;
+
+/**
+ * Add help for parent and all its children.
+ * Must be added to parent after all other commands.
+ */
+public class HelpCommand extends CommandTreeBase
+{
+    private final ICommand parent;
+
+    public HelpCommand(CommandTreeBase parent)
+    {
+        this.parent = parent;
+        for (ICommand command : parent.getSubCommands())
+        {
+            if (!this.equals(command))
+            {
+                addSubcommand(new HelpSubCommand(this, command));
+            }
+        }
+    }
+
+    @Override
+    public int getRequiredPermissionLevel()
+    {
+        return 0;
+    }
+
+    @Override
+    public String getName()
+    {
+        return "help";
+    }
+
+    @Override
+    public String getUsage(ICommandSender sender)
+    {
+        return String.format("/%s %s", parent.getName(), getName());
+    }
+
+    @Override
+    public void execute(MinecraftServer server, ICommandSender sender, String[] args) throws CommandException
+    {
+        if (args.length == 0)
+        {
+            sender.sendMessage(new TextComponentTranslation(parent.getUsage(sender)));
+            for (ICommand subCommand : getSubCommands())
+            {
+                if (subCommand instanceof HelpSubCommand && subCommand.checkPermission(server, sender))
+                {
+                    subCommand.execute(server, sender, args);
+                }
+            }
+            return;
+        }
+        super.execute(server, sender, args);
+    }
+
+    public static class HelpSubCommand extends CommandBase
+    {
+        private final HelpCommand parent;
+        private final ICommand command;
+
+        public HelpSubCommand(HelpCommand parent, ICommand command)
+        {
+            this.parent = parent;
+            this.command = command;
+        }
+
+        @Override
+        public int getRequiredPermissionLevel()
+        {
+            return 0;
+        }
+
+        @Override
+        public String getName()
+        {
+            return command.getName();
+        }
+
+        @Override
+        public String getUsage(ICommandSender sender)
+        {
+            return String.format("%s %s", parent.getUsage(sender), getName());
```

Comment:
`getUsage` returns a string to be localized. This technically works, but could probably be cleaner.

Reply:
This isn't really used, it was kind of just thrown in there so it wasn't blank.
Do you have an ideas for what could go here?